test/compilable/nodefaultlibs.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
```

Comment:
I know this may be an unpopular opinion, but please use D instead of shell scripts for these kinds of tests so they are platform independent.

Reply:
Would love to, can you explain how I can write this test in D?

Reply:
It would require importing Phobos, using the facilities in `std.process` to invoke the compiler, and then examining the output.  The test itself could be invoked with the host's `rdmd`.  Importing Phobos into the test suite may be controversial.  Others would have to speak up about that.  I don't have any problem with it, and would even encourage it.
The basic premise is that users are having a hell of a time building DMD, druntime, and phobos on Windows, and running the test suite.  Part of the problem is Windows users have to install a Posix-like environment to make it work and it's not even a natural fit for Windows.  D (through Phobos) abstracts most of the platform dependencies, so the same code works on all platforms that D supports.  If the tests and build scripts are written in D, it should work everywhere D is supported.

Reply:
Yeah I've brought up the issues with using bash on windows quite a bit and have received little to no response. I can't even get anyone to publish instructions on how to run the test suite on windows.

But for now it's what we have. Maybe I'll come up with an idea the leadership will respond to...but will have to be another PR.

Reply:
[build.d](https://github.com/dlang/dmd/blob/master/src/build.d) works today on Linux, Windows, and MacOS.  However, we need to migrate more of the makefile targets to it, and then we need to do the same for druntime and Phobos.  If that interests you and are willing to volunteer your skills to move it forward, it would be most welcome.  Precedent is already there, so it should be less of a hassle now.

Reply:
Here's one idea: https://github.com/dlang/dmd/pull/9848